ddtrace/tracer/textmap.go

```diff
@@ -114,21 +116,136 @@ func NewPropagator(cfg *PropagatorConfig) Propagator {
 	if cfg.PriorityHeader == "" {
 		cfg.PriorityHeader = DefaultPriorityHeader
 	}
-	return &propagator{cfg}
+	return &chainedPropagator{
+		injectors:  makeInjectors(cfg),
+		extractors: makeExtractors(cfg),
+	}
+}
+
+// chainedPropagator implements Propagator and applies a list of injectors and extractors.
+// When injecting, all injectors are called to propagate the span context.
+// When extracting, it tries each extractor, selecting the first successful one.
+type chainedPropagator struct {
+	injectors  []Propagator
+	extractors []Propagator
+}
+
+// makeInjectors returns a list of injectors to apply when propagating a
+// context. By default, only the datadog propagation style will be used.
+// If the DD_PROPAGATION_STYLE_INJECT environment variable is set,
+// this can override the default behavior.
+func makeInjectors(cfg *PropagatorConfig) []Propagator {
+	dd := &propagator{cfg}
+	b3 := &propagatorB3{}
+
+	ps := os.Getenv("DD_PROPAGATION_STYLE_INJECT")
+	if ps == "" {
+		return []Propagator{dd}
+	}
+	styles := propagationStyleList(ps)
+
+	var injectors []Propagator
+	for _, v := range styles {
+		switch strings.ToLower(v) {
+		case "datadog":
+			injectors = append(injectors, dd)
+		case "b3":
+			injectors = append(injectors, b3)
+		default:
+			// TODO(cgilmour): consider logging something for invalid/unknown styles.
+		}
+	}
+
+	if len(injectors) == 0 {
+		// If all the styles were invalid/unknown, then revert to default behavior.
+		return []Propagator{dd}
+	}
+	return injectors
 }
 
-// propagator implements a propagator which uses TextMap internally.
-// It propagates the trace and span IDs, as well as the baggage from the
-// context.
-type propagator struct{ cfg *PropagatorConfig }
+func makeExtractors(cfg *PropagatorConfig) []Propagator {
+	dd := &propagator{cfg}
+	b3 := &propagatorB3{}
+
+	ps := os.Getenv("DD_PROPAGATION_STYLE_EXTRACT")
+	if ps == "" {
+		return []Propagator{dd}
+	}
+	styles := propagationStyleList(ps)
+
+	var extractors []Propagator
+	for _, v := range styles {
+		switch strings.ToLower(v) {
+		case "datadog":
+			extractors = append(extractors, dd)
+		case "b3":
+			extractors = append(extractors, b3)
+		default:
+			// TODO(cgilmour): consider logging something for invalid/unknown styles.
+		}
+	}
+
+	if len(extractors) == 0 {
+		// If all the styles were invalid/unknown, then revert to default behavior.
+		return []Propagator{dd}
+	}
+	return extractors
+}
+
+// propagationStyleList splits a string containing propagation styles
+// separated by space or comma, and returns a []string containing
+// only the propagation styles.
+func propagationStyleList(s string) []string {
+	f := func(r rune) bool {
+		switch r {
+		case ' ', ',':
+			return true
+		}
+		return false
+	}
+	return strings.FieldsFunc(s, f)
```

Comment:
If you decide to keep this, let’s not have a separate function and just inline it:
```go
strings.FieldsFunc(s, func(r rune) bool { return r == ' ' || r == ',' })
```